org.scala-ide.sdt.core/src/org/scalaide/core/completion/ScalaCompletions.scala

```diff
@@ -25,11 +25,21 @@ import org.scalaide.util.internal.Commons
 class ScalaCompletions extends HasLogger {
   import org.eclipse.jface.text.IRegion
 
+  @deprecated("Use getCompletions, which correctly handles units with a different presentation compiler", since = "4.1.0")
+  def findCompletions(region: IRegion, position: Int, icu: InteractiveCompilationUnit): List[CompletionProposal] =
+    icu.scalaProject.presentationCompiler { compiler =>
+      findCompletion(region, position, icu, compiler)
+    }.getOrElse(Nil)
+
+
   /** Returns the list of possible completions, at the given position in the compilation unit,
    *  with the given region as completion prefix.
+   *
+   *  @note This supports compilation units that have a different presentation compiler than the default
+   *        per-project compiler
    */
-  def findCompletions(region: IRegion, position: Int, icu: InteractiveCompilationUnit): List[CompletionProposal] =
-    icu.scalaProject.presentationCompiler { compiler =>
+  def getCompletions(region: IRegion, position: Int, icu: InteractiveCompilationUnit): List[CompletionProposal] =
```

Comment:
The problem is the JVM: a trait becomes an interface and an implementation class. In this case:

```scala
interface IncrementalCompilationUnit {
..
  def presentationCompiler: IScalaPresentationCompilerProxy
}

class  IncrementalCompilationUnit$class {
  static def presentationCompiler($this: IncrementalCompilationUnit): IScalaPresentationCompilerProxy = // my implementation
}
```

every subclass of this trait will get a synthetic implementation for the interface method. For example, for play2:

```scala
class Play2CompilationUnit extends ... {
  // forwarder
  def presentationCompiler = IncrementalCompilationUnit$class.presentationCompiler(this)
}
```

So, changing the interface makes all subclasses inherit a new (abstract) method, and the forward will be missing. That's ok as long as nobody is calling that method, but in this PR I also add a *call* to the new method. So, unless subclasses recompile and republish, they'd be broken.